### PR TITLE
Do not add kir::TensorIndex::root_domain_index_map_

### PR DESCRIPTION
**Pull request by naoyam**
Just cleans up `kir::TensorIndex` by not adding the additional field.

### CODE REVIEW

**Comment by naoyam** [comment]
The main difference is that the getIndex functions are refactored so that all individual indices mapped onto the root domain are returned without creating TensorIndex. The vectorizer uses this list of indices to figure out the vector extent. Since all root indices are available, there is no need to modify TensorIndex. I think this is less intrusive.

File: torch/csrc/jit/codegen/cuda/lower_unroll.cpp [hunk 1]
```diff
@@ -176,25 +191,22 @@ kir::ForLoop* handleMisalignedVectorization(
...
 
   // Disable vectorize flag in child For-Loop
   loop_structure.back()->setVectorize(false);
 
-  auto address = ir_builder.create<kir::TensorIndex>(
-      index->view()->fuserTv(),
-      index->indices(),
-      index->rootDomainIndexMap(),
-      true);
+  auto address =
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/lower_unroll.cpp:227]: Don't we want the address of the global memory tensor? I think `non_vec_tv` is the non-global memory tv.

Comment (rdspring1) [reply torch/csrc/jit/codegen/cuda/lower_unroll.cpp:227]: Ah. Maybe `non_vec_tv` should be `global_tv`, and `vec_tv` => `local_vec_tv`

File: torch/csrc/jit/codegen/cuda/lower_unroll.cpp [hunk 2]
```diff
@@ -168,6 +182,7 @@ kir::ForLoop* handleMisalignedVectorization(
   auto in_tv = vec_expr->inputs().front()->as<kir::TensorView>();
   kir::TensorView* vec_tv =
       (out_tv->memoryType() != MemoryType::Global) ? out_tv : in_tv;
+  kir::TensorView* non_vec_tv = (vec_tv == out_tv) ? in_tv : vec_tv;
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/lower_unroll.cpp]: Maybe just check `out_tv->memoryType() == MemoryType::Global`. It seems more clear to me in hindsight.

**Review by rdspring1 — APPROVED** [review]
I think renaming `non_vec_tv` to `global_tv` will make the code clearer.

**Comment by naoyam** [comment]
Thanks. I updated the variable names and added more comments and checks.

**Pull request merged**